doc/architectural/static-functions.md

```diff
@@ -0,0 +1,153 @@
+\ingroup module_hidden
+\page modular-verification-static Modular Verification of Static Functions
+
+\section verification-of-static Modular Verification of Static Functions
+
+\author Kareem Khazem
+
+This page describes how to use CBMC on static functions.
+
+\tableofcontents
+
+CBMC can check libraries and other codebases that expose several
+entry points. To do this, users typically write a *harness* that
+captures the entry points' API contract, and that calls into the API
+with unconstrained values. The example below shows such a library and
+harness:
+
+\code{.c}
+void public_api_function(const int *a, int *b)
+{
+  // ...
+  private_function(a, b);
+  // ...
+}
+
+static void private_function(const int *a, int *b)
+{
+  // ...
+}
+\endcode
+
+The harness sets up some assumptions and then calls into the API:
+
+\code{.c}
+void harness()
+{
+  int a, b;
+  __CPROVER_assume(a < 10);
+  __CPROVER_assume(a >= 0);
+  public_api_function(&a, &b);
+  __CPROVER_assert(b != a);
+}
+\endcode
+
+The following commands build and check this function:
+
+\code{.sh}
+> goto-cc -c -o library.o library.c
+> goto-cc -c -o harness.o harness.c
+> goto-cc -o to_check.gb library.o harness.o
+> cbmc to_check.gb
+\endcode
+
+\subsection stubbing-out-static Stubbing out static functions
+
+For performance reasons, it might be desirable to analyze the static
+function separately from the API function. We can analyze the API
+function by "stubbing out" the static function, replacing it with a
+function that does nothing apart from asserting that its inputs satisfy
+the function's contract. Add the following to `harness.c`:
+
+\code{.c}
+static void private_function(const int *a, int *b)
+{
+  __CPROVER_assert( /* ... */ );
```

Comment:
Should the asserts and the assumes be equal? If so show that in your example

Reply:
Done, thanks